### ISSUE TITLE
virt-manager: '/usr/libexec/qemu-bridge-helper' is not a suitable bridge helper

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#34972]
## Issue description

I configured QEMU/KVM and connect virt-manager via "User session" to start a VM.

The VM has no network connection, so i try to set my ethernet interface as bridge. (I don't know what i'm doing)

![screenshot_2018-02-14_19-58-19](https://user-images.githubusercontent.com/91113/36222418-7fa7d88e-11c1-11e8-9023-ba1c1993ad5d.png)

Starting the VM fails now with this message:
```
Error starting domain: '/usr/libexec/qemu-bridge-helper' is not a suitable bridge helper: No such file or directory

Traceback (most recent call last):
  File "/nix/store/xv0z2k1cxi3jgp8kqmksq68gvi1n6sw1-virt-manager-1.4.2/share/virt-manager/virtManager/asyncjob.py", line 89, in cb_wrapper
    callback(asyncjob, *args, **kwargs)
  File "/nix/store/xv0z2k1cxi3jgp8kqmksq68gvi1n6sw1-virt-manager-1.4.2/share/virt-manager/virtManager/asyncjob.py", line 125, in tmpcb
    callback(*args, **kwargs)
  File "/nix/store/xv0z2k1cxi3jgp8kqmksq68gvi1n6sw1-virt-manager-1.4.2/share/virt-manager/virtManager/libvirtobject.py", line 82, in newfn
    ret = fn(self, *args, **kwargs)
  File "/nix/store/xv0z2k1cxi3jgp8kqmksq68gvi1n6sw1-virt-manager-1.4.2/share/virt-manager/virtManager/domain.py", line 1489, in startup
    self._backend.create()
  File "/nix/store/s3pr9pb9awj3b0fn1fgrlvqjvycycpqz-libvirt-python-3.6.0/lib/python2.7/site-packages/libvirt.py", line 1062, in create
    if ret == -1: raise libvirtError ('virDomainCreate() failed', dom=self)
libvirtError: '/usr/libexec/qemu-bridge-helper' is not a suitable bridge helper: No such file or directory
```

### Steps to reproduce

```
  # hypervisor virtualization
  boot.kernelModules = [ "kvm-intel" ];
  virtualisation.libvirtd.enable = true;
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.80, NixOS, 17.09.3001.395fe7f8bcb (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.11.16`
 - channels(davidak): `""`
 - channels(root): `"nixos-17.09.3001.395fe7f8bcb, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by volth** [comment]
To set your ethernet interface as bridge

```
networking = {
  bridges.br0.interfaces = [ "eth0" ];
  interfaces.br0.ip4 = [{ address = "1.2.3.4"; prefixLength = 24; }];  /* former settings of eth0 */
};
```

Then it should look in ```virt-manager``` like this:

![](http://i.imgur.com/yGKmWLU.png)

**Comment by davidak** [comment]
Thanks. Is `'/usr/libexec/qemu-bridge-helper' is not a suitable bridge helper: No such file or directory` an error that should be fixed?

**Comment by volth** [comment]
@davidak It could be fixed. But it is not easy (it is a suid executable and suid executables are not first class citizens in NixOS, they require extra wrapper) and not very useful (it performs ad hoc network configuration which NixOS are usually done declarative)

**Comment by davidak** [comment]
So we let the issue open as "known problem" until someone wants to fix it? Or is there anything we can do easily to have a better user experience (aka don't get cryptic error messages)?

**Comment by volth** [comment]
just to keep backlog of ```virt-manager```-on-NixOS problems (maybe I fix some of them eventually).
*  too high CPU consumption sometimes when it expected to be idle. Maybe a spin loop or something
* "shutdown" does not work with ```qemu``` guests. just no reaction
* "force off" of ```lxc``` can put the system into strange state which would require reboot (a easy fix would to disable "force off" for them)

**Comment by Baughn** [comment]
It's a problem, but what can we use instead?

*Not* having VMs isn't a useful alternative.

**Comment by nagisa** [comment]
Adding a

```
    virtualisation.libvirtd.qemuVerbatimConfig = ''
      bridge_helper = "${pkgs.qemu}/libexec/qemu-bridge-helper"
    ''
```

seems to not help at all. The file at `/var/lib/libvirtd/qemu.conf` does get the line in it, but libvirtd itself seems to ignore the configuration entirely and still insists on using `/usr/libexec/qemu-bridge-helper`

**Comment by atisharma** [comment]
Still a problem on 22.05.

**Comment by mfenniak** [comment]
This seems to be fixed in #187211, which I assume should be in upcoming NixOS 22.11.  It's working for me on unstable.
